### ISSUE TITLE
touchscreen rotate and calibration 

### ISSUE THREAD

**Issue by computeridea** [opencardev/crankshaft#94]
i, i'm tryng cranckshaft openauto with a custom display touchscreen, it work fine in EGL mode, but the axis are inverted, how can i change it?
for axis i means just the otuch axis, not the display video itself, that is fine as it is, i tried even to put 1 on flip_rotate in cranckshaft.sh but doesn't works
b.regards
Davide

**Comment by chunky74** [comment]
Not sure if this will help - but I had a similar problem with an overlay touchscreen.  My fix was to remove my ribbon connector and replace it in the other way around.

**Comment by computeridea** [comment]
exactly.. for start voice input i have to touch right-bottom corner in change of right-upper corner
if si move the touch horizzontal the screen rool upo and down, if i touch vertical it put away event widjet from screen

**Comment by chunky74** [comment]
Could you post a link to the touchscreen you are using please ?

**Comment by computeridea** [comment]
https://rover.ebay.com/rover/0/0/0?mpre=https%3A%2F%2Fwww.ebay.co.uk%2Fulk%2Fitm%2F262118994729

**Comment by Vitalic66** [comment]
Had similar problems with these capacitive touch. Bought an inductive on. No setup needed anymore...

You might change xinput settings. There are several instructions how to setup eGalax touchscreens. Guess you have one too...

**Comment by computeridea** [comment]
i have two touch controller,  one capacitive, but it doesn't works with raspberry or crankshaft, and one resistive, with the capacitive one it does as yous say if i cionnect directly on android, if i invert the cable, it inverts the axis, but it inverts boths axis, but what is needed is to rotate axis of 90°,  anyway the resistive one doesn't have effect.. and it's the one that works on raspberry.. 
so i think another solution is needed

**Comment by computeridea** [comment]
hi guyes, thanks to you all, in the end i could calibrate the eGalax controller, througt raspian and copied the configuration on crankshaft, but i have still a problem, in the demo screen, before i connect the phone, i can follow the touch curson on the screen for each corner, but when i connect the phone, it seems like the touch doesn't work on right and on botton, i can slide the widjet on center screen, but i can't push button on botton bar, or start voice command on upper right..
.. maybe it concern something about resolution when google auto start?  any advise?

**Comment by WhereisBielefeld** [comment]
Hey,
I´m struggling with the same issues already mentioned. I also have got a eGalax Controller (Sainsmart 7" Display) and got my values Touchscreen set in <del>"`/etc/X11/xorg.conf.d/99-calibration.conf`"</del> but no effect. Neither in AA nor in OpenAuto. I also varied the values but nothing changed after a reboot nor before.

Section "InputClass"
   Identifier "calibration"
   MatchProduct "eGalax Inc. USB TouchController"
   Option "Calibration" "121 1917 317 1741"
   Option "SwapAxes" "1"
EndSection

Is there a trick or did I overread something?
I started with X11 (set in Crankshaft config). Any clues?
